include/seastar/core/current_task.hh

```diff
@@ -0,0 +1,8 @@
+#pragma once
+
+#include <seastar/core/task.hh>
+
+namespace seastar {
+    task *current_task();
+}
```

Comment:
For me, this all should be in an `#ifdef`, and moved to a separate file indicating that the use is in deadlock detection.